### PR TITLE
[DX-1402] docs: add deposit withdrawal api guides

### PR DESCRIPTION
**Pull request by sprngbk**
## Description

This PR updates the deposit and withdrawal guides with API instructions.

## Resolved issues
Jira: https://immutable.atlassian.net/browse/DX-1402

### Before submitting the PR, please consider the following:
- [x] It's beneficial if your pull request references an issue used to discuss the pull request ahead of time. If you haven't previously created an issue, please create one and discuss your contribution with the maintainers.
- [x] Prefix your PR title with `feat: `, `fix: `, `chore: `, `docs:`, or `refactor:`.
- [x] The description should clearly illustrate what problems the pull request solves.
- [x] Ensure that the commit messages follow our guidelines.
- [x] Resolve merge conflicts (if any).
- [x] Make sure that the current branch is up to date with the `main` branch.

### For internal Immutable developers:
- [x] If you are adding or updating a documentation page, please also add or update the team ownership of that page (follow [this guide](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1947927045/Docs+ownership+registry#What-to-do-when-adding%2Fupdating-documentation-pages%3F)).
- [x] If you are adding or updating documentation for an SDK, please make sure that you meet the requirements in [this doc](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1916994017/SDK+documentation+guide).
- [ ] Please obtain PR approval from all of the following:
    - Your tech lead (Fallback: Team lead or engineering manager)
    - Your product manager (Fallback: Group PM)
    - Another software engineer on your team

### CODE REVIEW

File: docs/main/guides/basic-guides/deposits-withdrawals/index.md [hunk 1]
```diff
@@ -186,4 +187,227 @@ The function below will deposit ETH from L1 to L2. If you wish to deposit a diff
...
+    console.log(response);
+}).catch(err => {
+    console.error(err);
+})
+```
+
+#### 4. Call contract to complete withdrawal
+
+```js
+import { Contracts, Config } from '@imtbl/core-sdk';
```
Comment (zatlite) [inline docs/main/guides/basic-guides/deposits-withdrawals/index.md:396]: if you're importing core-sdk why not just use the completeWithdrawal method from the SDK?

File: docs/main/guides/basic-guides/deposits-withdrawals/index.md [hunk 2]
```diff
@@ -186,4 +187,227 @@ The function below will deposit ETH from L1 to L2. If you wish to deposit a diff
...
+// Get instance of core contract
+const contract = Contracts.Core.connect(
+  config.ethConfiguration.coreContractAddress,
+  ethSigner,
+);
+
+// Populate and send transaction
+const populatedTransaction = await contract.populateTransaction.withdraw(
+  starkPublicKey, // Use `stark_key` obtained in previous step
+  assetType, // "ETH", "ERC20" or "ERC721"
```
Comment (zatlite) [inline docs/main/guides/basic-guides/deposits-withdrawals/index.md:409]: This sample code doesn't show how to get assetType. Maybe it is better to use the sdk's completeWithdrawal method?

File: docs/main/guides/basic-guides/deposits-withdrawals/index.md (hunk 1 above)
Comment (sprngbk) [reply docs/main/guides/basic-guides/deposits-withdrawals/index.md:396]: That's a good point - I think we're just using the SDK to initialize the contract in the example, but people can manually call it and follow our subsequent API example if they want to.

File: docs/main/guides/basic-guides/deposits-withdrawals/index.md (hunk 2 above)
Comment (sprngbk) [reply docs/main/guides/basic-guides/deposits-withdrawals/index.md]: I just updated it with instructions on retrieving this value

File: docs/main/guides/basic-guides/deposits-withdrawals/index.md [hunk 3]
```diff
@@ -390,7 +390,48 @@ fetch("https://api.sandbox.x.immutable.com/v1/withdrawals", {
...
+
+const encodeAssetBody = {
+  token: {
+    data: {
+      blueprint: "string",
+      id: "string",
+      token_address: "string",
+      token_id: "string"
+    },
+    type: "ETH"
```
Comment (zatlite) [inline docs/main/guides/basic-guides/deposits-withdrawals/index.md:409]: or ERC20 or ERC721

**Review by zatlite — APPROVED** [review]

**Pull request merged**